dll/cpl/input/input_list.c

```diff
@@ -323,14 +323,56 @@ InputList_AddInputMethodToUserRegistry(DWORD dwIndex, INPUT_LIST_NODE *pNode)
         }
     }
 
+    /* 
+     * First time more then 1 input language exists switching hotkeys needed
+     * set default values for language/layout switching hotkeys
+     */
+    if (2 == dwIndex) {
+        if (RegOpenKeyExW(HKEY_CURRENT_USER,
+                      L"Keyboard Layout\\Toggle",
+                        0,
+                        KEY_SET_VALUE,
+                        &hKey) == ERROR_SUCCESS)
+        {
+            WCHAR szBuffer[MAX_PATH];
+
+            /* 'Hotkey' and 'Layout Hotkey' have always the same value */
+            wcscpy(szBuffer, L"1");
+
+            RegSetValueExW(hKey,
+                        L"Hotkey",
+                        0,
+                        REG_SZ,
+                        (LPBYTE)szBuffer,
+                        (wcslen(szBuffer) + 1) * sizeof(WCHAR));
+
+            RegSetValueExW(hKey,
+                        L"Language Hotkey",
+                        0,
+                        REG_SZ,
+                        (LPBYTE)szBuffer,
+                        (wcslen(szBuffer) + 1) * sizeof(WCHAR));
+
+            wcscpy(szBuffer, L"2");
+
+            RegSetValueExW(hKey,
+                        L"Layout Hotkey",
+                        0,
+                        REG_SZ,
+                        (LPBYTE)szBuffer,
+                        (wcslen(szBuffer) + 1) * sizeof(WCHAR));
+
+            RegCloseKey(hKey);
+        }
+    }
```

Comment:
If this code block is part of the workaround you mentioned in the JIRA report, please mention that explicitly in the code!

Reply:
No it is not, workaround is in a separate patch only attached to JIRA